upt_macports/upt_macports.py

```diff
@@ -28,6 +29,408 @@ def _render_makefile_template(self):
+                    "SugarCRM-1.1.3": "unknown",
+                    "TAPR-OHL-1.0": "unknown",
+                    "TCL": "unknown",
+                    "TCP-wrappers": "unknown",
+                    "TMate": "unknown",
+                    "TORQUE-1.1": "unknown",
+                    "TOSL": "unknown",
+                    "TU-Berlin-1.0": "unknown",
+                    "TU-Berlin-2.0": "unknown",
+                    "UPL-1.0": "unknown",
+                    "Unicode-DFS-2015": "unknown",
+                    "Unicode-DFS-2016": "unknown",
+                    "Unicode-TOU": "unknown",
+                    "Unlicense": "unknown",
+                    "VOSTROM": "unknown",
+                    "VSL-1.0": "unknown",
+                    "Vim": "unknown",
+                    "W3C": "unknown",
+                    "W3C-19980720": "unknown",
+                    "W3C-20150513": "unknown",
+                    "WTFPL": "unknown",
+                    "Watcom-1.0": "unknown",
+                    "Wsuipa": "unknown",
+                    "X11": "unknown",
+                    "XFree86-1.1": "unknown",
+                    "XSkat": "unknown",
+                    "Xerox": "unknown",
+                    "Xnet": "unknown",
+                    "YPL-1.0": "unknown",
+                    "YPL-1.1": "unknown",
+                    "ZPL-1.1": "unknown",
+                    "ZPL-2.0": "unknown",
+                    "ZPL-2.1": "unknown",
+                    "Zed": "unknown",
+                    "Zend-2.0": "unknown",
+                    "Zimbra-1.3": "unknown",
+                    "Zimbra-1.4": "unknown",
+                    "Zlib": "unknown",
+                    "bzip2-1.0.5": "unknown",
+                    "bzip2-1.0.6": "unknown",
+                    "copyleft-next-0.3.0": "unknown",
+                    "copyleft-next-0.3.1": "unknown",
+                    "curl": "unknown",
+                    "diffmark": "unknown",
+                    "dvipdfm": "unknown",
+                    "eCos-2.0": "unknown",
+                    "eGenix": "unknown",
+                    "gSOAP-1.3b": "unknown",
+                    "gnuplot": "unknown",
+                    "iMatix": "unknown",
+                    "libpng-2.0": "unknown",
+                    "libtiff": "unknown",
+                    "mpich2": "unknown",
+                    "psfrag": "unknown",
+                    "psutils": "unknown",
+                    "wxWindows": "unknown",
+                    "xinetd": "unknown",
+                    "xpp": "unknown",
+                    "zlib-acknowledgement": "unknown"
+                }
+        try:
+            macports_license = licenses[spdx_license]
+        except:
+            macports_license = "unknown"
+        return macports_license
+
+    def _licences(self):
+        for i in self.upt_pkg.licenses:
+            i.name = self.spdx_to_macports(i.spdx_identifier)
+        return self.upt_pkg.licenses
```

Comment:
When used as a loop index, "i" usually refers to a number (as in "for i in range(42)"). Use a more explicit name, such as "license". It is also a bit weird to modify self.upt_pkg.licenses. This could become:

> return [self.spdx_to_macports(license.spdx_identifier) for license in self.upt_pkg.licenses]